docs/book/src/tasks/experimental-features/cluster-class/change-clusterclass.md

```diff
@@ -1,32 +1,192 @@
+When planning a change to a ClusterClass, users should always take into consideration
+how those changes might impact the existing Clusters already using the ClusterClass, if any.
+
+There are two strategies for defining how a ClusterClass change rolls out to existing Clusters: 
+
+- Roll out ClusterClass changes to existing Cluster in a controlled/incremental fashion.
+- Roll out ClusterClass changes to all the existing Cluster immediately.
+
+The first strategy is the recommended choice for people starting with ClusterClass; it
+requires the users to create a new ClusterClass with the expected changes, and then
+[rebase](#Rebase) each Cluster to use the newly created ClusterClass.
+
+By splitting into separated step the change to the ClusterClass change and its roll-out
+on the Clusters the user will reduce the risk of introducing unexpected changes on the
+existing Clusters, or at least limit the blast radius of those changes
+to a small number of Clusters already rebased (in fact it is similar to a canary deployment).
+
+The second strategy listed above instead requires to change a ClusterClass "in place", which can
+be simpler and faster that creating a new ClusterClass, but this approach
+implies that changes are immediately propagated to all the Clusters already using the 
+ClusterClass being changed.
+
+While this is powerful, users should be well aware of the risk that an operation impacting
+many Clusters and the same time imply and how this could percolate to the underlying infrastructure
+in case the operation triggers machines roll out across the entire fleet of Clusters.
+
+However, no matter of which strategy you are choosing for implementing your changes to a ClusterClass, 
+please make sure to:
+
+- Understand what [Compatibility Checks](#Compatibility Checks) are and how to prevent changes
+  that can lead to non-functional Clusters.
+- [Dry running ClusterClass changes](#Dry running ClusterClass changes) before applying them.
+
+If instead you are interested in understanding more about how which kind of  
+effects you should expect on the Clusters, or if you are interested in additional details 
+about the internals of the topology reconciler you can start reading the notes in the 
+[dry run](#Dry running ClusterClass changes) documentation or looking at the [reference](#Reference)
+documentation at the end of this page.
+
+## Changing ClusterClass templates
+
+Templates are an integral part of a Cluster Class, and thus the same considerations
+described in the previous paragraph applies; that's means that also when you are changing
+a template referenced in a ClusterClass users should always plan for how the
+change should be propagated to the existing Clusters and choose the strategy that best
+suite the expectation.
+
+Then, according to the [Cluster API operational practices](https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html),
```

Comment:
Can we make this a relative link? (so that it also works on versioned versions of the book)